components/ws-manager/pkg/manager/annotations.go

```diff
@@ -108,6 +108,9 @@ func (m *Manager) markWorkspace(ctx context.Context, workspaceID string, annotat
 	err := retry.RetryOnConflict(backoff, func() error {
 		pod, err := m.findWorkspacePod(ctx, workspaceID)
 		if err != nil {
+			if isKubernetesObjNotFoundError(err) {
+				return nil
+			}
 			return fmt.Errorf("cannot find workspace %s: %w", workspaceID, err)
```

Comment:
if we filter not found error above, then error message below no longer makes sense.
```suggestion
			return fmt.Errorf("cannot mark workspace %s: %w", workspaceID, err)
```

Reply:
fixed